### DISCUSSION TITLE
Mouse orientation with rotation_angle option

### DISCUSSION THREAD

**Discussion by quincerly** [BlitterStudio/amiberry#1497] [Q&A] [answered]
I've been trying to get Amiberry working with a Pimoroni Hyperpixel 4 GPIO lcd and current version of PiOS (Bookworm 32bit) without using X. I need to rotate the display clockwise 90deg to get the correct orientation. Using the rotation_angle option in the cfg file does correctly rotate the display for the GUI but the mouse controls are unrotated and so it is unusable. Is there a way to rotate the mouse axes too? I'm trying with V5.7.4.

**Comment by midwan** [comment]
You are correct, this is a bug. When it was implemented, it was only tested with games, so I missed this scenario. 

Should be a rather easy fix though, will check.

**Reply by midwan** [reply, marked as answer]
This should be fixed now, in the latest `master`. It will eventually be back-ported to Amiberry-Lite as well.

**Reply by quincerly** [reply]
I've just tried the 7.0.0 RC 1 deb and it doesn't seem to be fixed there. Issue #1521.

**Comment by quincerly** [comment]
Brilliant. It’s currently very slowly building on my Pi 3A+…………..zzzz….. Will let you know how I do.

**Reply by quincerly** [reply]
I'm not having much luck - probably my incompetence. I can build on my Pi 3B from amiberry master and amiberry-lite master. Both run but very slowly - the mouse pointer in the GUI is very jerky even thought it's quite smooth if I use the distributed v5.7.4 binaries on the same machine. I'm also failing to get even the screen to rotate in my built versions - perhap I'm not setting the config right, but I tried an amiberry.cfg which does rotate the screen on v5.7.4 binaries. Until now I've been using a very old build of Amiberry on ancient Raspbian (Buster I think) as it was working fine for me, but recent death of my old Hyperpixel finally forced an update. Any idea where I might be going wrong?

**Reply by midwan** [reply]
Amiberry-Lite hasn't been updated with the changes from the main Amiberry yet. That will happen after the next release is out, since I want to keep the merges to a minimum.
The rotation changes are currently _only_ in Amiberry, not the Lite version. However, this version is more demanding as it uses the latest WinUAE emulation code, it will not run full speed on anything less than a Pi5 in most cases.

It might be good enough to test the results, but I expect you would be better off waiting for Amiberry-Lite to be updated after that.

**Reply by quincerly** [reply]
Thanks. Yes, I understood that the fix in not in the Lite version yet. My problem is that that both the full and lite versions when I compile them on my Pi 3B are very slow compared to your downloadable v5.7.4 binaries and a the super-smooth V3.0 I've been using on an ancient version of Raspbian, so I wondered if I'm compiling wrong. I'm following the instructions on the GitHub README for each.
